configure

```diff
@@ -347,7 +347,7 @@ if test "$gcc" -eq 1 && ($cc $CFLAGS -c $test.c) >> configure.log 2>&1; then
     CFLAGS="${CFLAGS} ${ARCHS}"
     LDFLAGS="${LDFLAGS} ${ARCHS}"
   fi
-  CFLAGS="${CFLAGS} -Wall"
+  CFLAGS="${CFLAGS} -Wall -Werror=implicit-function-declaration"
```

Comment:
Why is this necessary?

Reply:
It basically forbids compilation if a function prototype was not found before. It's a very common flag to have, Linux kernel does it too.

Also, see commit 8943aa3b6f012da9386a8347a7f6dd24debb9e82.

Reply:
In theory this is a good idea as only in C++ prototypes are actually required... In C, any undefined function is expected to return `int`, which might for example mask truncation if the actual return type is larger.

Reply:
Not all compilers may support this flag, so you gotta test for it.

Reply:
This really should be in all sane C compilers by now. It's first introduced in GCC in 1997: https://github.com/gcc-mirror/gcc/blob/releases/gcc-2.8.0/gcc/ChangeLog#L1868

The Linux kernel also unconditionally enforces it on all compilers.

But if the maintainers insist, I'm find guarding this with $gcc.

Reply:
> The Linux kernel also unconditionally enforces it on all compilers.

There aren't many compilers in that list ;)

See for example https://github.com/spack/spack/issues/23830. Test don't assume... It's easy enough.

Reply:
It would have to have a test to make sure it actually works.
You could also split that into a separate PR, that way we could merge the softfp abi fix.

Reply:
> It would have to have a test to make sure it actually works. You could also split that into a separate PR, that way we could merge the softfp abi fix.

If we are not going to support LTO with configure, the softfp ABI fix will be redundant and unnecessary.

Reply:
Ok, then this PR can be focused on just adding the function declaration error.